packages/web/src/pages/chat-page/ChatPage.tsx

```diff
@@ -1,28 +1,86 @@
-import { chatSelectors, FeatureFlags, useProxySelector } from '@audius/common'
+import { useCallback, useEffect, useRef } from 'react'
+
+import {
+  chatSelectors,
+  FeatureFlags,
+  Status,
+  useProxySelector
+} from '@audius/common'
+import { push as pushRoute } from 'connected-react-router'
+import { useDispatch } from 'react-redux'
 import { RouteComponentProps } from 'react-router-dom'
+import useMeasure from 'react-use-measure'
 
+import { useSelector } from 'common/hooks/useSelector'
 import Page from 'components/page/Page'
 import { useFlag } from 'hooks/useRemoteConfig'
+import { chatPage } from 'utils/route'
 
 import styles from './ChatPage.module.css'
 import { ChatComposer } from './components/ChatComposer'
 import { ChatHeader } from './components/ChatHeader'
 import { ChatList } from './components/ChatList'
 import { ChatMessageList } from './components/ChatMessageList'
 
-const { getOtherChatUsers } = chatSelectors
+const { getOtherChatUsers, getChatMessagesStatus } = chatSelectors
 
 const messages = {
   messages: 'Messages'
 }
 
 export const ChatPage = ({ match }: RouteComponentProps<{ id?: string }>) => {
-  const chatId = match.params.id
+  const currentChatId = match.params.id
+  const dispatch = useDispatch()
   const { isEnabled: isChatEnabled } = useFlag(FeatureFlags.CHAT_ENABLED)
   const users = useProxySelector(
-    (state) => getOtherChatUsers(state, chatId),
-    [chatId]
+    (state) => getOtherChatUsers(state, currentChatId),
+    [currentChatId]
+  )
+  const messagesStatus = useSelector((state) =>
+    currentChatId ? getChatMessagesStatus(state, currentChatId) : Status.IDLE
   )
+
+  // Get the height of the header so we can slide the messages list underneath it for the blur effect
+  const [headerRef, bounds] = useMeasure({
+    polyfill: ResizeObserver,
+    offsetSize: true
+  })
+  const messagesRef = useRef<HTMLDivElement>(null)
+
+  // Cache whether the user was already scrolled to the bottom
+  // Effects run after the render, so we can't calculate this inside the effect
+  // but without rendering, there's nowhere to scroll, so do that within the effect
+  let wasAtBottom = false
+  if (messagesRef.current) {
+    const { scrollTop, clientHeight, scrollHeight } = messagesRef.current
+    wasAtBottom = scrollTop + clientHeight >= scrollHeight
+  }
```

Comment:
maybe pull this out into a standalone hook?

Reply:
will think about it. it's the only place it's used now, so leaning towards no

Reply:
👍 imo nice to not have low level concerns like height measurements interspersed with higher level logic, DRY aside